roles/setup/tasks/main.yml

```diff
@@ -119,62 +119,77 @@
-    - '/tmp/hccm_openshift_usage_scheduled_report.yaml'
-  when:
+- name: Define temp report file data location
+  tempfile:
+    state: directory
+    suffix: 'korekuta-{{ ocp_cluster_id }}'
+  register: 'ocp_temp_file_dir'
+
+# tag block
+- block:
+    - name: Switch to Operator Metering project
+      command: '{{ OCP_CLI }} project {{ OCP_METERING_NAMESPACE }}'
+      register: oc_project
+      changed_when: false
+
+    - name: Check for hccm-usage-report Report
+      command: '{{ OCP_CLI }} get Report {{ setup_report_name }}'
+      register: oc_get_report
+      changed_when: false
+      ignore_errors: true
+      when:
+        - oc_project.stdout is defined and oc_project != ''
+
+    - name: Check for hccm-usage-report ReportGenerationQuery
+      command: '{{ OCP_CLI }} get ReportGenerationQuery {{ setup_report_name }}'
+      register: oc_get_report_generation_query
+      changed_when: false
+      ignore_errors: true
+      when:
+        - oc_project.stdout is defined and oc_project != ''
+
+    - name: Check for hccm-openshift-persistentvolumeclaim Report
+      command: '{{ OCP_CLI }} get Report {{ setup_storage_report_name }}'
+      register: oc_get_storage_report
+      changed_when: false
+      ignore_errors: true
+      when:
+        - oc_project.stdout is defined and oc_project != ''
+
+    - name: Check for hccm-openshift-persistentvolumeclaim ReportGenerationQuery
+      command: '{{ OCP_CLI }} get ReportGenerationQuery {{ setup_storage_report_name }}'
+      register: oc_get_storage_report_generation_query
+      changed_when: false
+      ignore_errors: true
+      when:
+        - oc_project.stdout is defined and oc_project != ''
+
+    - name: Copy hccm-usage-report files
+      copy:
+        src: '{{ item }}'
+        dest: '{{ ocp_temp_file_dir }}'
+      with_fileglob:
+        - 'files/kube*'
+        - 'files/pod*'
+        - 'files/hccm*'
+        - 'files/persistentvolume*'
+      when:
+        - oc_project.stdout is defined and oc_project != ''
+        - oc_get_report.rc is defined and  oc_get_report.rc != 0
+        - oc_get_storage_report.rc is defined and  oc_get_storage_report.rc != 0
+        - oc_get_report_generation_query.rc is defined and  oc_get_report_generation_query.rc != 0
+        - oc_get_storage_report_generation_query.rc is defined and  oc_get_storage_report_generation_query.rc != 0
+
+    - name: Create Operator Metering definitions
+      command: '{{ OCP_CLI }} create -f {{ ocp_temp_file_dir }}'
+      when:
+        - oc_project.stdout is defined and oc_project != ''
+        - oc_get_report.rc is defined and  oc_get_report.rc != 0
+        - oc_get_storage_report.rc is defined and  oc_get_storage_report.rc != 0
+        - oc_get_report_generation_query.rc is defined and  oc_get_report_generation_query.rc != 0
+        - oc_get_storage_report_generation_query.rc is defined and  oc_get_storage_report_generation_query.rc != 0
+  tags:
     - setup_reports
```

Comment:
Think the tag should just be `- setup` per the playbook: https://github.com/project-koku/korekuta/blob/master/ocp_usage_playbook.yml#L8